sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -22,17 +50,224 @@ class TableCheckpointStore:
+            dic[u'sequence_number'] = entity[u'sequence_number']
+            dic[u'offset'] = str(entity[u'offset'])
+            checkpointslist.append(dic)
+        return checkpointslist
 
-    def list_ownership(self, namespace, eventhub, consumergroup, **kwargs):
-        pass
+    def update_checkpoint(self, checkpoint):
+        """Updates the checkpoint using the given information for the offset, associated partition and
+        consumer group in the storage table.
+        Note: If you plan to implement a custom checkpoint store with the intention of running between
+        cross-language EventHubs SDKs, it is recommended to persist the offset value as an string.
+        :param Dict[str,Any] checkpoint: A dict containing checkpoint information:
+                - `fully_qualified_namespace` (str): The fully qualified namespace that the Event Hub belongs to.
+                  The format is like "<namespace>.servicebus.windows.net".
+                - `eventhub_name` (str): The name of the specific Event Hub the checkpoint is associated with,
+                  relative to the Event Hubs namespace that contains it.
+                - `consumer_group` (str): The name of the consumer group the checkpoint is associated with.
+                - `partition_id` (str): The partition ID which the checkpoint is created for.
+                - `sequence_number` (int): The sequence number of the :class:`EventData<azure.eventhub.EventData>`
+                  the new checkpoint will be associated with.
+                - `offset` (str): The offset of the :class:`EventData<azure.eventhub.EventData>`
+                  the new checkpoint will be associated with.
+        :rtype: None
+        """
+        checkpoint_entity = self._create_checkpoint_entity(checkpoint)
+        try:
+            self.table_client.update_entity(mode=UpdateMode.REPLACE, entity=checkpoint_entity)
+        except ResourceNotFoundError:
+            self.table_client.create_entity(entity=checkpoint_entity)
 
-    def list_checkpoints(self, namespace, eventhub, consumergroup, **kwargs):
-        pass
+    def _update_ownership(self, ownership):
+        ownership_entity = self._create_ownership_entity(ownership)
+        try:
+            if ownership['etag'] is None:
+                metadata = self.table_client.create_entity(entity=ownership_entity,
+                headers={'Prefer': 'return-content'})
+                ownership['etag'] = metadata['etag']
+                ownership['last_modified_time'] = _to_timestamp(dateutil.parser.isoparse
+                (metadata['content']['Timestamp']))
+                return ownership
+            metadata = self.table_client.update_entity(mode=UpdateMode.REPLACE, entity=ownership_entity,
+            etag=ownership['etag'], match_condition=MatchConditions.IfNotModified)
+            ownership['etag'] = metadata['etag']
+            entities = self.table_client.get_entity(partition_key=ownership_entity['PartitionKey']
```

Comment:
You're only getting a single entity, so maybe this could be renamed to `updated_entity`.

Reply:
make sense